polkadot-host-spec/c02-state.tm

```diff
@@ -476,42 +476,193 @@
+
+    \;
+
+    Keys that were modified outside of extrinsics, such as changes made by
+    the Runtime directly, have a <verbatim|0xffffffff> value as its indice.
+
+    \;
+
+    The Changes Trie is generated when the Runtime calls
+    <verbatim|ext_storage_changes_root> as described in section
+    <reference|sect-ext-storage-changes-root>.
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-blocks>The <strong|storage key to blocks
+    mappings> track any changes which occured in a certain range of blocks.
+    Its value is a SCALE encoded array containing block numbers where
+    extrinsics caused any changes to its key. The block numbers are
+    represented as unsigned 32-bit integers.
+
+    \;
+
+    The Changes Trie is generated when the Runtime calls
+    <verbatim|ext_storage_changes_root> as described in section
+    <reference|sect-ext-storage-changes-root>, but unlike the other mappings,
+    they are not generated on each block. The Runtime sets the key
+    <verbatim|:changes_trie> in storage with its value being a tuple of two
+    32-bit integers:
+
+    <\itemize-dot>
+      <item><verbatim|digest_interval> - The interval (in blocks) at which
+      block mappings are created. Block mappings are not created when this is
+      less or equal to <verbatim|1>.
+
+      <item><verbatim|digest_levels> - Maximal number of \Plevels\Q in the
+      hierarchy. <verbatim|0> means that block mappings are not created at
+      all. <verbatim|1> means only the regular <verbatim|digest_interval>
+      block mappings are created. Any other level means that the block
+      mappings are created every <verbatim|<math|digest_interval<rsup|level>>>
+      block for each level in <verbatim|1> to <verbatim|digest_levels>.
+    </itemize-dot>
+
+    The Polkadot node must track the interval and levels, verify if
+    conditions apply and then generate those mappings when the Runtime calls
+    <verbatim|ext_storage_changes_root>. The range spans from the last block
+    (where this kind of mappings were created) + 1 to the current block.
+
+    \;
+
+    Whenever the interval and level stored in <verbatim|:changes_trie>
+    changes the current interval is reset and an new digest according to the
+    changed parameters is started.
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-child-tries>The <strong|storage key to child
+    changes tries mappings> track any storage keys from child storages and
+    their corresponding Child Changes Trie. As described in Definition
+    <reference|defn-storage-key-to-extrinsics>, changes inside child storages
+    are inserted into their own Trie. The changed key is inserted into the
+    main Trie, where its value is the Merkle proof as desribed in section
+    <reference|sect-merkl-proof> of the corresponding Child Storage Trie,
+    which stores the mappings between the storage key and extrinsics.
+  </definition>
+
```

Comment:
We don't really talk about blocks before chapter 3. Shouldn't the following paragraph be moved to to where we spec the header of the block and then refer back to this section?

Reply:
As discussed, the Changes Trie section was moved to chapter 3.